[docs][PLAT-11684] Instructions for creating common TServer service (#20283)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md

## Connect to the universe

For information on how to connect to the universe from the Kubernetes cluster, as well as remotely, see [Connect YugabyteDB clusters](../../../deploy/kubernetes/clients/#connect-tls-secured-yugabytedb-cluster-deployed-by-helm-charts).

### Create common TServer service for zones

> By default each zone gets its own TServer service which can be used to connect to the universe. Optionally, you can create a common service across these zones as follows. 

Reviewer: ```suggestion
By default, each zone has its own YB-TServer service, and you can use this service to connect to the universe. Optionally, you can create a common service across all zones as follows. 
```

> By default each zone gets its own TServer service which can be used to connect to the universe. Optionally, you can create a common service across these zones as follows. 

Reviewer: If you create the common service, do the individual ones remain? What is the benefit?

Author: Mentioned that it is an additional service. 

About the benefits, I'm not sure how to write those here:
When you have a single service, your clients can connect to it and don't have to worry about individual TServers from different zones. If some zone's TServer(s) are down the request will be served by other zones TServer as the service takes care of load balancing. 

For now I have modified the sentence to say we create an highly available service with these steps.

Note that this requires all the zone deployments to be in the same namespace.

> 1. Set the following Kubernetes overrides to add universe name label on the TServer pods. You can do this during universe creation or by modifying the Kubernetes overrides of an existing universe.

Reviewer 2: Can we link to the overrides doc section here?

> 1. Set the following Kubernetes overrides to add universe name label on the TServer pods. You can do this [during universe creation](#configure-helm-overrides) or by [modifying the Kubernetes overrides](../../manage-deployments/edit-helm-overrides/) of an existing universe.

Reviewer: ```suggestion
1. Set the following Kubernetes overrides to add the universe-name label on the TServer pods. You can do this when you [create the universe](#configure-helm-overrides) or by [modifying the Kubernetes overrides](../../manage-deployments/edit-helm-overrides/) of an existing universe.

```

   ```yaml
   tserver:
     podLabels:
       universe-name: yb-k8s
   ```

> 1. Save following block as `yb-tserver-common-service.yaml` file. You can modify the service type, annotations, and the label selector according to your needs.

Reviewer: Where does the file go? Is it somehow assumed?

Author: We use this temporary file to save the block, make any modifications, and then apply it in the next step. 
After that the we don't care what happens with the file basically.

> 1. Save following block as `yb-tserver-common-service.yaml` file. You can modify the service type, annotations, and the label selector according to your needs.

Reviewer: ```suggestion

1. Save the following to a file named `yb-tserver-common-service.yaml`. You can customize the service type, annotations, and the label selector as required.

```

   ```yaml
   # yb-tserver-common-service.yaml
   apiVersion: v1
   kind: Service
   metadata:
     name: yb-k8s-common-tserver
     labels:
       app.kubernetes.io/name: yb-tserver
     # annotations:
     #   networking.gke.io/load-balancer-type: "Internal"
   spec:
     type: ClusterIP
     selector:
       app.kubernetes.io/name: yb-tserver
       # This value should match with the value from step 1.
       universe-name: yb-k8s
     ports:
     # Modify the ports if using non-standard ports.
     - name: tcp-yedis-port
       port: 6379
     - name: tcp-yql-port
       port: 9042
     - name: tcp-ysql-port
       port: 5433
   ```

> 1. Run the following command to create the service in the universe's namespace which is `yb-prod-yb-k8s` in our case.

Reviewer: ```suggestion

1. Run the following command to create the service in the universe's namespace (`yb-prod-yb-k8s` in this example).

```

   ```sh
   $ kubectl apply -f yb-tserver-common-service.yaml -n yb-prod-yb-k8s
   ```

> 1. You can access the YugabyteDB at `yb-k8s-common-tserver.yb-prod-yb-k8s.svc.cluster.local`.

Reviewer: ```suggestion

After the overrides are applied, in this example you would access the universe at `yb-k8s-common-tserver.yb-prod-yb-k8s.svc.cluster.local`.

```
